ObjectivePGP/ObjectivePGPObject.m

```diff
@@ -277,7 +277,13 @@ + (nullable NSData *)encrypt:(NSData *)dataToEncrypt addSignature:(BOOL)shouldSi
     NSData *content;
     if (shouldSign) {
         // sign data if requested
-        content = [self sign:dataToEncrypt detached:NO usingKeys:keys passphraseForKey:passphraseForKeyBlock error:error];
+        let signKeys = [NSMutableArray<PGPKey *> array];
+        for (PGPKey* key in keys){
+            if (key.secretKey && key.signingSecretKey != NULL){
+                [signKeys addObject:key];
+            }
+        }
+        content = [self sign:dataToEncrypt detached:NO usingKeys:signKeys passphraseForKey:passphraseForKeyBlock error:error];
```

Comment:
the filtering is more suited inside `-[ObjectivePGP sign:detached:usingKeys:passphraseForKey:]`